### PR TITLE
Make the React-based Markup view functional for Replay 

### PR DESCRIPTION
**Pull request by gabrielluong**
This requires turning on the pref in src/devtools/client/inspector/prefs.js
pref("devtools.inspector.features.new-markupview.enabled", false);

### CODE REVIEW

File: src/devtools/client/inspector/markup/new-markup.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+   * Expands the given node in the markup tree.
+   *
+   * @param  {String} nodeId
+   *         The actor ID of the node to expand.
+   */
+  async expandNode(nodeId) {
+    await this.getChildren(this.nodes.get(nodeId));
+    this.tree[nodeId].isExpanded = true;
+    this.store.dispatch(updateTree(this.tree));
+  }
```
Comment (jasonLaster) [inline src/devtools/client/inspector/markup/new-markup.js:149]: I think if it were me, I'd move getChildren and some of this logic to redux actions, but up to you.

File: src/devtools/client/inspector/inspector.js [hunk 2]
```diff
@@ -903,6 +907,13 @@ class Inspector {
       });
     }
 
+    if (features.newMarkupView) {
+      sidebarPanels.push({
+        id: "markupview",
+        title: INSPECTOR_L10N.getStr("inspector.panelLabel.markupView"),
+      });
+    }
```
Comment (jasonLaster) [inline src/devtools/client/inspector/inspector.js:915]: I wonder if we should just swap markup views when this is on?  Up to you.

**Review by jasonLaster — COMMENTED** [review]
Love this!

File: src/devtools/client/inspector/inspector.js (hunk 2 above)
Comment (gabrielluong) [reply src/devtools/client/inspector/inspector.js:915]: I am gonna work on that next.

File: src/devtools/client/inspector/markup/components/Node.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.onExpanderToggle = this.onExpanderToggle.bind(this);
+    this.onSelectNodeClick = this.onSelectNodeClick.bind(this);
+  }
+
+  onExpanderToggle(event) {
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply src/devtools/client/inspector/markup/components/Node.js:34]: I think we can say this change for later

**Comment by jasonLaster** [comment]
keyboard focus is weird, but that is just because there are two of the views

Noticed some errors creeping in.

Screen Shot 2020-09-22 at 4 34 26 PM

**Review by jasonLaster — APPROVED** [review]
Ship it

**Comment by gabrielluong** [comment]
We haven't implemented keyboard shortkey handling yet (that's next), and when we do, we won't have to worry about that tree walker anymore.

**Pull request merged**